jerry-core/jerry.cpp

```diff
@@ -1340,13 +1340,6 @@ jerry_init (jerry_flag_t flags) /**< combination of Jerry flags */
     JERRY_WARNING_MSG ("Ignoring memory statistics option because of '!MEM_STATS' build configuration.\n");
 #endif /* !MEM_STATS */
   }
-  else if (flags & (JERRY_FLAG_MEM_STATS_PER_OPCODE | JERRY_FLAG_MEM_STATS_SEPARATE))
-  {
-    flags &= ~(JERRY_FLAG_MEM_STATS_PER_OPCODE | JERRY_FLAG_MEM_STATS_SEPARATE);
-
-    JERRY_WARNING_MSG (
-      "Ignoring detailed memory statistics options because memory statistics dump mode is not enabled.\n");
```

Comment:
@lemmaa, the `else if` block checks that `--mem-stats-separate` and `--mem-stats-per-opcode` options are not passed if `--mem-stats` is not passed, i.e.  "separate" and "per-opcode" memory statistics mode are turned off without `--mem-stats` option.

Currently, behaviour in case `JERRY_FLAG_MEM_STATS_SEPARATE` flag is set, and `JERRY_FLAG_MEM_STATS` is not set - dump of statistics only for parser phase, and not for execution phase (the first of the dumps is performed in `jerry_parse`, and the second should be performed in in `mem_finalize`). So, in the case, only parser phase statistics are dumped, and not separate statistics for parser and execution phases.

 `JERRY_FLAGS_MEM_STATS` could be set in the `if` block, so implicitly enabling `--mem-stats` mode, if `JERRY_FLAG_MEM_STATS_SEPARATE` or `JERRY_FLAG_MEM_STATS_PER_OPCODE` are set.